src/server/api/controllers/vessel.controller.js

```diff
@@ -0,0 +1,116 @@
+"use strict";
+
+const HttpError = require("../lib/utils/http-error");
+const knex = require("../../config/db");
+const bcrypt = require("bcrypt");
+const generatePassword = require("generate-password");
+const mailer = require("../lib/services/mailer.service");
+const { hashPassword } = require("../lib/utils/hash.password");
+//GET
+const getVessels = async req => {
+  try {
+    return await knex("vessels").select(
+      "vessels.id",
+      "vessels.organization_id",
+      "vessels.mmsi",
+      "vessels.name"
+    );
+  } catch (err) {
+    console.log(err);
+  }
+};
+
+//GET:id
+const getVesselById = async id => {
+  try {
+    const vessels = await knex("vessels")
+      .select("vessels.id as id", "organization_id", "mmsi", "name")
+      .where({ id: id });
+    if (vessels.length === 0) {
+      throw new HttpError("Bad request", `Cannot find user for ID ${id}!`, 404);
+    }
+    return vessels;
+  } catch (err) {
+    return err.message;
+  }
+};
+//POST
+
+const createVessel = async ({ body }) => {
+  const { id, organization_id, mmsi, name } = body;
+
+  const organization = await knex
+    .from("organizations")
+    .select("*")
+    .where({
+      id: organization_id
+    });
+  if (organization.length === 0) {
+    throw new HttpError(
+      "Bad request",
+      `${organization_id} organization not found!`,
+      404
+    );
+  }
+
+  const vessels = await knex("vessels").insert({
+    organization_id: organization[0].id,
+    mmsi: mmsi,
+    name: name
+  });
+  return vessels;
+};
+
+//PATCH
+const editVessel = async ({ body }) => {
+  const { id, organization_id, mmsi, name } = body;
+  console.log(body);
+  const vessel = await knex
+    .from("vessels")
+    .select("*")
+    .where({
+      name: name
```

Comment:
You need to select by `ID`.

Right now you try to find vessels by matching the name that is passed into the parameters with what you have in the DB. But if I make a PATCH request trying to rename the vessel those two will not match.

So change this to ID and test it by sending a PATCH request with a body like this:
```
{
	"id": 1,
	"organization_id": 1,
	"mmsi": "414223",
	"name": "Fancy new name"
}
```

Reply:
find vessels by matching the name instead of id